subprojects/core/src/integTest/groovy/org/gradle/api/tasks/WritePropertiesIntegrationTest.groovy

```diff
@@ -45,32 +46,48 @@ class WritePropertiesIntegrationTest extends AbstractIntegrationSpec {
         """
 
         when:
-        succeeds "props"
+        runProps()
         then:
         file("output.properties").text == normalize("""
             #Line comment
             """)
     }
 
-    def "simple properties are written sorted alphabetically"() {
+    private runProps() {
+        result = expectOutputDeprecation(executer.withTasks("props")).run()
+    }
+
+    private expectOutputDeprecation(GradleExecuter runWithTasks) {
+        runWithTasks.expectDocumentedDeprecationWarning("he WriteProperties_Decorated.outputFile property has been deprecated. This is scheduled to be removed in Gradle 9.0. " +
+            "Please use the destinationFile property instead. See https://docs.gradle.org/current/dsl/org.gradle.api.tasks.WriteProperties_Decorated.html#org.gradle.api.tasks.WriteProperties_Decorated:outputFile for more details.")
```

Comment:
:x: The documentation link is invalid. It should point to `https://docs.gradle.org/current/dsl/org.gradle.api.tasks.WriteProperties.html` without the `_Decorated`